src/routes.js

```diff
@@ -0,0 +1,94 @@
+const Apify = require('apify');
+const querystring = require('querystring');
+
+const { utils: { log, requestAsBrowser } } = Apify;
+
+const { cleanProject, getToken, notifyAboutMaxResults } = require('./utils');
+const { BASE_URL, MAX_PAGES, PROJECTS_PER_PAGE } = require('./consts');
+
+exports.handleStart = async ({ request, session }, query, requestQueue, proxyConfig, maxResults) => {
+    // on this phase - getting TOKEN AND COOKIES
+    const { seed, cookies } = await getToken(request.url, session, proxyConfig);
+
+    const page = 1;
+    const totalProjects = 0;
+    const savedProjects = 0;
+    const maximumResults = Math.min(maxResults, MAX_PAGES * PROJECTS_PER_PAGE);
+
+    const params = querystring.stringify({
+        ...query,
+        page,
+        seed,
+    });
+    const listUrl = `${BASE_URL}${params}`;
+
+    // ADDING TO THE QUEUE FIRST PAGINATION PAGE WITH JSON
+    await requestQueue.addRequest({
+        url: listUrl,
+        userData: {
+            cookies,
+            page,
+            label: 'PAGINATION-LIST',
+            totalProjects,
+            savedProjects,
+            maximumResults,
+        },
+    });
+};
+
+exports.handlePagination = async ({ request, session }, requestQueue, proxyConfiguration) => {
+    let { page, totalProjects, savedProjects } = request.userData;
+    const { cookies, maximumResults } = request.userData;
+
+    // MAKING REQUEST => JSON OBJECT IN RESPONSE
+    const { body } = await requestAsBrowser({
+        url: request.url,
+        proxyUrl: proxyConfiguration.newUrl(session.id),
+        headers: {
+            Accept: 'application/json, text/javascript, */*; q=0.01',
+            'X-Requested-With': 'XMLHttpRequest',
+            Cookie: cookies,
+        },
+        json: true,
+    });
+
+    // ON THE FIRST PAGE WE ARE CHECKING IF WE REACHED THE LIMIT
+    if (page === 1) {
+        log.info(`Page ${page}: Found ${body.total_hits} projects in total.`);
+        // If kickstarter contains more then 2400 results for current query, notify user
+        // that he will not have all results and that he needs to refine his query.
+        if (body.total_hits > maximumResults) notifyAboutMaxResults(body.total_hits, maximumResults);
+        totalProjects = Math.min(body.total_hits, maximumResults);
+    }
+    // ARRAY OF THE PROJECTS FROM THE PAGE
+    log.info(`Number of  saved projects: ${savedProjects}`);
+    const projectsToSave = body.projects.slice(0, maximumResults - savedProjects)
```

Comment:
this might crash with a `cannot call slice of undefined`. catch this and re-throw a meaningful error to the user like `throw new Error('The page didn\'t load as expected, retrying...')`. this was an error that was happening the last version